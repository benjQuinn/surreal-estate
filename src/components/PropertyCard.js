import React from "react";
import "../styles/property-card.css";
import PropType from "prop-types";
import { GiCastle } from "react-icons/gi";
import { FaBath, FaBed, FaPoundSign, FaRegEnvelope } from "react-icons/fa";

const PropertyCard = ({ property }) => {
  const { title, type, city, bedrooms, bathrooms, price } = property;
  const formattedPrice = price.toLocaleString("en-GB", {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  });

  return (
    <div className="property-card">
      <div className="property-card-img">
        <GiCastle className="navbar-logo-icon" data-testid="navbar-logo-icon" />
      </div>
      <ul className="property-card-items">
        <li className="property-card-item">{title}</li>
        <li className="property-card-item">
          {type} - {city}
        </li>
        <li className="property-card-item">
          <FaBed className="propert-card-icon" />
          {bedrooms}
        </li>
        <li className="property-card-item">
          <FaBath className="propert-card-icon" />
          {bathrooms}
        </li>
        <li className="property-card-item">
          <FaPoundSign className="propert-card-icon" />
          {formattedPrice}
        </li>
      </ul>
      <button className="property-card-email-button" type="submit">
        <FaRegEnvelope className="propert-card-icon" />
        Email
      </button>
    </div>
  );
};

PropertyCard.propTypes = {
  property: PropType.shape({
    title: PropType.string.isRequired,
    type: PropType.string.isRequired,
    bathrooms: PropType.number.isRequired,
    bedrooms: PropType.number.isRequired,
    price: PropType.number.isRequired,
    city: PropType.string.isRequired,
    email: PropType.string.isRequired,
  }).isRequired,
};

export default PropertyCard;
