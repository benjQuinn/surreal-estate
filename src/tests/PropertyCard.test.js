import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "5 bedroom house",
    type: "Detached",
    bedrooms: 5,
    bathrooms: 2,
    city: "Manchester",
    price: 500000,
    email: "qwerty@email.com",
  };

  it("renders correctly", () => {
    const { asFragment } = render(<PropertyCard {...validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders prop values correctly", () => {
    const { getByText } = render(<PropertyCard {...validProps} />);

    expect(getByText(/5 bedroom house/i).textContent).toBe(validProps.title);
    expect(getByText(/detached - manchester/i).textContent).toBe(
      `${validProps.type} - ${validProps.city}`
    );
    expect(getByText(5).textContent).toBe(validProps.bedrooms.toString());
    expect(getByText(2).textContent).toBe(validProps.bathrooms.toString());
    expect(getByText(/500,000.00/).textContent).toBe(
      validProps.price.toLocaleString("en-GB", {
        minimumFractionDigits: 2,
        maximumFractionDigits: 2,
      })
    );
  });
});
