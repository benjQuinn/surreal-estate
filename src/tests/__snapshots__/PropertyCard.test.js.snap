// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="property-card-img"
    >
      <svg
        class="navbar-logo-icon"
        data-testid="navbar-logo-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M255.95 27.11L180.6 107.614l150.7 1.168-75.35-81.674h-.003zM25 109.895v68.01l19.412 25.99h71.06l19.528-26v-68h-14v15.995h-18v-15.994H89v15.995H71v-15.994H57v15.995H39v-15.994H25zm352 0v68l19.527 26h71.06L487 177.906v-68.01h-14v15.995h-18v-15.994h-14v15.995h-18v-15.994h-14v15.995h-18v-15.994h-14zm-176 15.877V260.89h110V126.63l-110-.857zm55 20.118c8 0 16 4 16 12v32h-32v-32c0-8 8-12 16-12zM41 221.897V484.89h78V221.897H41zm352 0V484.89h78V221.897h-78zM56 241.89c4 0 8 4 8 12v32H48v-32c0-8 4-12 8-12zm400 0c4 0 8 4 8 12v32h-16v-32c0-8 4-12 8-12zm-303 37v23h-16v183h87v-55c0-24 16-36 32-36s32 12 32 36v55h87v-183h-16v-23h-14v23h-18v-23h-14v23h-18v-23h-14v23h-18v-23h-14v23h-18v-23h-14v23h-18v-23h-14v23h-18v-23h-14zm-49 43c4 0 8 4 8 12v32H96v-32c0-8 4-12 8-12zm72 0c8 0 16 4 16 12v32h-32v-32c0-8 8-12 16-12zm80 0c8 0 16 4 16 12v32h-32v-32c0-8 8-12 16-12zm80 0c8 0 16 4 16 12v32h-32v-32c0-8 8-12 16-12zm72 0c4 0 8 4 8 12v32h-16v-32c0-8 4-12 8-12zm-352 64c4 0 8 4 8 12v32H48v-32c0-8 4-12 8-12zm400 0c4 0 8 4 8 12v32h-16v-32c0-8 4-12 8-12z"
        />
      </svg>
    </div>
    <ul
      class="property-card-items"
    >
      <li
        class="property-card-item"
      >
        5 bedroom house
      </li>
      <li
        class="property-card-item"
      >
        Detached - Manchester
      </li>
      <li
        class="property-card-item"
      >
        <svg
          class="propert-card-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 640 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
          />
        </svg>
        5
      </li>
      <li
        class="property-card-item"
      >
        <svg
          class="propert-card-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
          />
        </svg>
        2
      </li>
      <li
        class="property-card-item"
      >
        <svg
          class="propert-card-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M308 352h-45.495c-6.627 0-12 5.373-12 12v50.848H128V288h84c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-84v-63.556c0-32.266 24.562-57.086 61.792-57.086 23.658 0 45.878 11.505 57.652 18.849 5.151 3.213 11.888 2.051 15.688-2.685l28.493-35.513c4.233-5.276 3.279-13.005-2.119-17.081C273.124 54.56 236.576 32 187.931 32 106.026 32 48 84.742 48 157.961V224H20c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h28v128H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h296c6.627 0 12-5.373 12-12V364c0-6.627-5.373-12-12-12z"
          />
        </svg>
        500,000.00
      </li>
    </ul>
    <button
      class="property-card-email-button"
      type="submit"
    >
      <svg
        class="propert-card-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
        />
      </svg>
      Email
    </button>
  </div>
</DocumentFragment>
`;
